File: ServerSide/BallPassSchedulePattern/Players_QrcodeReaderTypes/Decode_e2_BitDataProcessorPlayerDir/BitDataProcessorPlayer.py
import os, sys
sys.path.append("../..")
from Players_CommonPlayers.SuperPlayerDir.SuperPlayer import SuperPlayer

class BitDataProcessorPlayer(SuperPlayer):
    def __init__(self):
        super().__init__()
        self.my_name = None  # 必ずNoneで初期化

    def return_my_name(self):
        return "BitDataProcessorPlayer"

    def process_bit_data(self, bit_data):
        """
        1次元リストで受け取ったビットデータを8ビットごとに区切って、
        新しいリストとして返すメソッド。
        
        入力: [0001011010111101010110] のようなビットデータ
        出力: 8ビットごとに区切ったリスト ['00010110', '10111101', ...]
        """
        # 8ビットごとにスライスして新しいリストに格納
        return [bit_data[i:i+8] for i in range(0, len(bit_data), 8)]

    def main(self):
        """
        リストデータを取得して、それを8ビットごとに
        分割し、戻り値として返す。
        """
        """ 入力 """
        woP = self.one_time_world_instance.rightBottomReaderPlayer
        self.replaced_matrix = woP.replaced_matrix # 置き換え後のmatrix
        self.png_file_path = woP.png_file_path # 写真のパス
        self.new_list2d = woP.new_list2d # データ部分をn*2のリストに変換したもの
        self.data_read = woP.data_read # データ部分を1次元リスト化したもの。

        
        # 仮のビットデータを用意
        bit_data = "0001011010111101010110"  # 22ビットのデータ
        
        # ビットデータを8ビットごとに変換
        processed_data = self.process_bit_data(bit_data)
        
        # 結果を表示
        print(processed_data)

        # 自身を更新して、return "Completed"を返す
        self.one_time_world_instance.bitDataProcessorPlayer = self
        return "Completed"

File: InstanceSide/BallPassSchedulePattern/Players_QrcodeReaderTypes/Decode_e_BitDataProcessorPlayerDir/BitDataProcessorPlayer.cs
using UdonSharp;
using UnityEngine;

public class BitDataProcessorPlayer : SuperPlayer
{
    public string pngFilePath; // 写真のパス
    public string dataRead; // データ部分を1次元リスト化したもの
    public string[] processedData; // 8ビット毎にスプリットしたデータ

    public RinaNumpy rinaNumpy; // RinaNumpyインスタンスをアタッチ

    void Start()
    {
        ResetPlayer(); // 初期化
    }

    public void ResetPlayer()
    {
        myName = "BitDataProcessorPlayer"; // プレイヤー名を設定
        pngFilePath = ""; // 初期化
        dataRead = ""; // 初期化
        processedData = new string[0]; // 初期化
    }

    public override string ReturnMyName()
    {
        return "BitDataProcessorPlayer";
    }

    private string[] ProcessBitData(string bitData)
    {
        /*
         * RinaNumpyを活用してビットデータを8ビットごとに分割
         */
        if (string.IsNullOrEmpty(bitData))
        {
            Debug.LogError("Input bitData is null or empty.");
            return new string[0];
        }

        int dataLength = bitData.Length;
        int splitCount = Mathf.CeilToInt(dataLength / 8f); // 8ビットごとに分割した個数を計算
        string[] result = new string[splitCount];

        for (int i = 0; i < splitCount; i++)
        {
            int startIndex = i * 8;
            int length = Mathf.Min(8, dataLength - startIndex);
            result[i] = bitData.Substring(startIndex, length); // RinaNumpyはここで必要なし
        }

        return result;
    }

    public override string ExecuteMain()
    {
        /*
         * データを取得して8ビットごとに分割し、結果を保存する
         */
        RightBottomReaderPlayer woP = (RightBottomReaderPlayer)oneTimeWorldInstance.GetComponent(typeof(RightBottomReaderPlayer));

        if (woP == null)
        {
            Debug.LogError("RightBottomReaderPlayer is not attached.");
            return "Error";
        }

        pngFilePath = woP.pngFilePath; // 写真のパスを取得
        dataRead = woP.dataRead; // データ部分を取得

        if (string.IsNullOrEmpty(dataRead))
        {
            Debug.LogError("DataRead is null or empty.");
            return "Error";
        }

        // RinaNumpyを使った処理
        processedData = ProcessBitData(dataRead);

        // 自身を更新
        BitDataProcessorPlayer updatedPlayer = (BitDataProcessorPlayer)oneTimeWorldInstance.GetComponent(typeof(BitDataProcessorPlayer));
        if (updatedPlayer != null)
        {
            updatedPlayer.pngFilePath = pngFilePath;
            updatedPlayer.dataRead = dataRead;
            updatedPlayer.processedData = processedData;
        }

        Debug.Log("BitDataProcessorPlayer processing completed with RinaNumpy.");
        return "Completed";
    }
}
